src/Sentry/GlobalSessionManager.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.IO;
+using System.Threading;
+using Sentry.Extensibility;
+using Sentry.Internal;
+
+namespace Sentry
+{
+    // AKA client mode
+    internal class GlobalSessionManager : ISessionManager
+    {
+        private readonly object _lock = new();
+        private readonly SentryOptions _options;
+
+        private string? _cachedInstallationId;
+        private Session? _currentSession;
+
+        public GlobalSessionManager(SentryOptions options)
+        {
+            _options = options;
+        }
+
+        private string GetInstallationId()
+        {
+            // Prevent race conditions that could cause us to generate multiple IDs.
+            // Note: this has to be synchronized across multiple processes too.
+            lock (_lock)
+            {
+                // Avoid IO if possible
+                if (!string.IsNullOrWhiteSpace(_cachedInstallationId))
+                {
+                    return _cachedInstallationId;
+                }
+
+                var filePath = Path.Combine(
+                    Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
+                    "Sentry",
+                    ".installation"
+                );
+
+                var directoryPath = Path.GetDirectoryName(filePath);
+                if (!string.IsNullOrWhiteSpace(directoryPath))
+                {
+                    Directory.CreateDirectory(directoryPath);
+                }
+
+                // Try to read existing
+                try
+                {
+                    return _cachedInstallationId = File.ReadAllText(filePath);
+                }
+                catch (FileNotFoundException)
+                {
+                }
```

Comment:
Would also be nice to have logging here to know we failed to read the file. Also that we're creating the directory.

And generally some error handling on the other I/O. CreateDirectory above would be the first to throw.
Today I spent one hour looking at a customer crash. That's the time yo appreciate good logs.